### PR TITLE
Introduce deprecation warnings for renaming module interpolation to interpolate and move make_equidistant from misc.tools into this new module

### LINKED ISSUE #133: Transfer make_equidistant to new interpolation module
We originally introduced `make_equidistant()` just as a part of _misc.tools_ but now we should move it into the new module _uncertainty.interpolation_.

### PR DESCRIPTION
**Pull request by BjoernLudwigPTB**
Prepares to resolve #133 and #134.

### CODE REVIEW

**Comment by BjoernLudwigPTB** [comment]
> We should consider also merging the tests `test_interpolation.py` and `test_make_equidistant` into the new `test_interpolate.py`. (Because it is now the same module and probably can share many of the hypothesis strategies.)

This should be resolved by 359e768ef45e7d51435be8606737cfcc436d17eb and 959d5cd5aa38a7a16a8738b21082699f65bcc43c.

File: test/test_interpolate.py [hunk 1]
```diff
@@ -30,37 +31,44 @@ def timestamps_values_uncertainties_kind(
...
+            Same as fill_value, but just for the uncertainties. (default is None)
         returnC : bool, optional
-            If True we request the sensitivities to be returned. If False we do not
-            request them. Defaults to False.
+            If True we request the sensitivities to be returned. If False (default) we
+            do not request them.
+        for_make_equidistant : bool, optional
+            If True we return the expected parameters for calling `make_equidistant()`.
+            If False (default) we return the expected parameters for calling
+            `make_equidistant()`.
```
Comment (mgrub) [inline test/test_interpolate.py:71]: If False --> `interp1d_unc()` ?

Comment (BjoernLudwigPTB) [reply test/test_interpolate.py:71]: Oh yes!

Comment (BjoernLudwigPTB) [reply test/test_interpolate.py]: Thanks for the pointer!

**Pull request merged**